### PR TITLE
Fix another FOR correlation bug I introduced earlier

### PR DESCRIPTION
**Pull request by msullivan**

### CODE REVIEW

File: edb/edgeql/compiler/stmt.py [hunk 1]
```diff
@@ -128,8 +128,18 @@ def compile_ForQuery(
...
+        # correlation between the iterator and the enclosing
+        # query, since the correlation imposes compilation issues
+        # we aren't willing to tackle.
+        if contains_dml:
+            ictx.path_scope.factoring_allowlist.update(
+                ctx.iterator_path_ids)
         iterator_view = stmtctx.declare_view(
             iterator, qlstmt.iterator_alias,
+            factoring_fence=contains_dml,
+            temporary=False,
```
Comment (elprans) [inline edb/edgeql/compiler/stmt.py:142]: The underlying cause seems to be a bug in `attach_subtree`.  Removing `temporary` here helps in this particular case, but the bug is still there.  Here's an example that's broken and doesn't use `FOR`:

```
WITH MODULE test
SELECT count(
    (Card.name, (SELECT (SELECT Card)))
)
```

Comment (msullivan) [reply edb/edgeql/compiler/stmt.py:142]: Per slack discussion, the real broken example is
```
        WITH MODULE test
        SELECT (
            Card.name,
            (WITH X := (SELECT Card) SELECT X.name),
        )
```

**Comment by 1st1** [comment]
Hm, is this PR stale?

**Comment by msullivan** [comment]
I'm closing this because I'm shortly submitting a PR that takes this approach (don't use `temporary` in `declare_view`) and applies it to all uses of declare_view, which fixes Elvis's issue as well.

**Pull request closed without merging**